pkgs/ffigen/test/header_parser_tests/expected_bindings/_expected_enum_int_mimic_bindings.dart

```diff
@@ -0,0 +1,45 @@
+// AUTO GENERATED FILE, DO NOT EDIT.
+//
+// Generated by `package:ffigen`.
+// ignore_for_file: type=lint
+import 'dart:ffi' as ffi;
+
+abstract class Simple {
+  static const int A0 = 0;
+  static const int A1 = 1;
+}
+
+abstract class SimpleWithNegative {
+  static const int B0 = 0;
+  static const int B1 = -1000;
+}
+
+abstract class ExplicitType {
+  static const int E0 = 0;
+  static const int E1 = 1;
+}
+
+final class Test extends ffi.Struct {
+  @ffi.UnsignedInt()
```

Comment:
Does this work as expected? Don't we need https://github.com/dart-lang/native/pull/1187#issuecomment-2145634534?

Maybe add a test with `2^31+42` as enum value so that you exercise the fact that it's uint32 instead of int32.

Reply:
Added a test for this, but looks like windows will always overflow, but on other OSs we the compiler will only overflow if there are no other negative enums
Basically you can't trust the values to be correct either, once you go beyond INT32_MAX.

```C
enum PositiveIntOverflow {
    C0=INT32_MAX + 42L,
};
enum NoIntOverflow {
    C1=-1,
    C2=INT32_MAX + 42L,
};
enum NegativeIntOverflow {
    C3=INT32_MIN - 42L,
};
```

becomes

```dart
enum PositiveIntOverflow { // @ffi.Int()
  C0(-2147483607);
}

enum NoIntOverflow { // @ffi.Long()
  C1(-1),
  C2(2147483689); // We need to set ignore-source-errors for windows tests to run, which then is -2147483607
}

enum NegativeIntOverflow { // @ffi.Long()
  C3(-2147483690);
}
```

Reply:
Looks like the compiler will always wrap around the signed MAX limit of the type

```C
enum ExplicitTypeWithOverflow : uint16_t {
    F1=INT16_MAX + 42,
};
```

becomes

```dart
enum ExplicitTypeWithOverflow {
  F1(-32727);
}
```

Reply:
Okay that's super finicky behavior then. Maybe just throw a warning on enum values outside the declared integer type and a warning on enums with longs.

Or leave it as is, peeps will have issues in their normal native code without FFI and FFIgen if they have such enums. So it's probably only people who dev their native code together with Dart and FFIgen that would ever be helped by such error messages.

Reply:
> Or leave it as is,

This seems best to me tbh. I've removed the troubling cases from the tests.